flink-table/flink-table-planner/src/main/java/org/apache/flink/table/expressions/DistinctAgg.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.expressions;
+
+/**
+ * The distinct aggregate expression.
+ */
+public class DistinctAgg extends UnaryExpression {
```

Comment:
Yes, we can using the `call`, but `DistinctAgg1 is created to prevent the user from continuously calling distinct, for example: `a.count.distinct.distinct`.

Reply:
I think it again, `Call` is work, but for user friendly, we can using the `AggregateCall`. i.e: Add AggregateCallExpression to avoid distinct on CallExpression directly which would bring chaos on Api. This is also consistent with the previous Flink master.

Reply:
IMHO I think we should allow `a.count.distinct.distinct` and fail during validation. Because this is also the behavior of the Java API, right? Right now, you can also do weird things like `a.count.trim.abs`. The API does not limit this as well.

Reply:
Yes，we can not avoid the wrong pattern in Java API, we only have the chance to fail during validation.  I really change the `distinctAgg` to the `Call`, but i find that current master code, the Scala API do not allow the follows pattern:
1. multi distinct, e.g., 'a.sum.distinct.distinct
2. wrong order, e.g., 'a.distinct.sum
3. distinct on udf, e.g., 'a.udf.distinct
So, we add the `AggregateCallExpression` for Scala user. 
What do you think?